app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -125,27 +125,29 @@ impl List {
         }
     }
 
-    /// Build the list, sorting all group lists and groups' contents appropriately.
-    pub fn build(self) -> component::List {
+    /// Build the list, sorting all group lists and groups' contents appropriately. Set the
+    /// favorites in the list keeping their order as passed.
+    pub fn build_with_favorites(self, favorites: component::group::List) -> component::List {
         for group in self.module_groups.values() {
             group.content.update_sorting("");
             if let Some(flattened) = &group.flattened_content {
                 flattened.update_sorting("");
             }
         }
         let top_modules_iter = self.module_groups.values().filter(|g| g.is_top_module);
-        let mut top_mdl_bld = component::group::ListBuilder::default();
+        let mut top_mdl_bld = component::group::AlphabeticalListBuilder::default();
         top_mdl_bld.extend(top_modules_iter.clone().map(|g| g.content.clone_ref()));
-        let mut top_mdl_flat_bld = component::group::ListBuilder::default();
+        let mut top_mdl_flat_bld = component::group::AlphabeticalListBuilder::default();
         top_mdl_flat_bld.extend(top_modules_iter.filter_map(|g| g.flattened_content.clone()));
         component::List {
-            all_components:        Rc::new(self.all_components),
-            top_modules:           top_mdl_bld.build(),
+            all_components: Rc::new(self.all_components),
+            top_modules: top_mdl_bld.build(),
             top_modules_flattened: top_mdl_flat_bld.build(),
-            module_groups:         Rc::new(
+            module_groups: Rc::new(
                 self.module_groups.into_iter().map(|(id, group)| (id, group.build())).collect(),
             ),
-            filtered:              default(),
+            filtered: default(),
+            favorites,
```

Comment:
The components from favorites section shall be also put in "all components" list, because we iterate over it when updating filtering. Also, the groups "update filtering" methods should be also called (I don't see a code for that).

Reply:
Ouch, I completely forgot about filtering - thanks for catching this! ❤️ 

Please note that the favorites list will be filtered, but not sorted. I believe keeping the order of components in a favorites list is reasonable from a UX point of view, reinforcing a user's memory of the order of components in the virtual group as designed by the library author. From the implementation point of view, this is notably easier to do, as the library author's order of components would have to be restored from some extra storage after in-place sorting/filtering is disabled.